### DISCUSSION TITLE
About frozened model test error

### DISCUSSION THREAD

**Discussion by shuaijiang-ustc** [deepmodeling/deepmd-kit#268] [Q&A]
Hi, there!

After frozening the model, I intend to test the model using 'dp test', got the error message:
dpmd120-cpu/lib/python3.7/site-packages/deepmd/calculator.py", line 52, in <listcomp>
atype = [self.type_dict[k] for k in symbols]
KeyError: 'S'

Locate this calculator.py file and print self.dp.get_type_map(), the output of self.dp.get_type_map() is empty!

Here is my situation: multiple systems are trained altogther. Under each folder, there is one system including type.raw and type_map.raw and the 'type_raw' settting in json file is removed. The multiple system training is perfectly good accroding to the RMSE of energy and force. However, based on the frozen model, the pb file, 'dp test' as well as some tests through dp coupling with ASE shows the same error message above.

Any suggestions are sinerely welcomed!

Shuai

**Comment by njzjz** [comment]
Your pb file does not store elements when training.
https://github.com/deepmodeling/deepmd-kit/blob/59d780c77aaf0c0692d1bb56a630c50bc987ea13/examples/water/train/water_se_a.json#L3-L4

You can also pass `type_dict` to `DP` class, e.g. `{"H":0, "O":1}`.
https://github.com/deepmodeling/deepmd-kit/blob/59d780c77aaf0c0692d1bb56a630c50bc987ea13/source/train/calculator.py#L31-L35

**Comment by shuaijiang-ustc** [comment]
Many thanks for the quick and helpful reply!
The first method through add "type_map" back into json file and re-train the model is working.
For the second mothod, I add this in calculator.py:
            if self.dp.get_type_map():
                self.type_dict = dict(zip(self.dp.get_type_map(), range(self.dp.get_ntypes())))
            else:
                type_map_list=["O", "H", "S", "C", "N"]
                self.type_dict=dict(zip(type_map_list, range(self.dp.get_ntypes())))
The result reports error: [1]+  Floating point exception.
Is this error arsing from training parameters settings?
Thanks.
